### ISSUE TITLE
Fresh install requires several imports from Bunq

### ISSUE THREAD

**Issue by colinerd** [firefly-iii/firefly-iii#1895]
**Bug description**
I am running Firefly III version 4.7.9 - dev

Running a fresh install on 4.7.9 - dev (updated from 4.7.8 release)
I have set up the Bunq import via API and ran the update. After it was finished, the balances were totally off and I was missing transactions. I had to run the import 5 times before it did not process any more transactions.

In my case, I started using Bunq since August 2017.

**Steps to reproduce**
Fresh Install of Firefly iii
Set up Bunq Import with API key
Run the import
Review the results - balances are completely off and transactions are missing that are explaining the differences in balances.

**Expected behavior**
Run the import and all transactions are imported (although it might take a bit longer to import)

**Extra info**
Is there a timeout on the import that it might not import all transactions on the first import?

**Comment by JC5** [comment]
Yes, this is a known issue. The next version of Firefly III will be better at importing large data sets.

More importantly: are you still missing transactions? I know this happens, but I have not been able to pinpoint the exact cause. I believe it may have to do with DUO and/or other government deposits.

**Comment by colinerd** [comment]
Difficult to say as I have lots of them :-)
After just running the import for 5 times, I came down to a difference of less then 50 euro’s per account on 3 of them, one of them is caused by the starting balance on the account. 2 accounts were on the spot. Running the import again did not seem to add any new transactions anymore.

What I can do in the next week, if I can find some time, is run through them and see if I can find anything in particular and report back here on my findings. I do not have DUO but do receive a monthly return on my tax income.

**Comment by colinerd** [comment]
I noticed I am missing one of my automatic saving transfers of today. I have 6 today but only see 5 in Firefly and did ran the import multiple times. Newer automatic saving transfers are also visible but this one appears missing. 
It is a transaction that matches the amount and date with another one of today however, I am only seeing one of them.

Could the import be ignoring a transaction that might appear the same but is not?

**Comment by JC5** [comment]
Yes, it is designed to. But the api from bunq should add unique markers. I can test this, stay tuned.

**Comment by colinerd** [comment]
Since I am curious as well :-)
Just did 2 transactions of 4 euro’s, should send 1 euro to the saving goal. Ran the import 3 times, last 2 times there were no new transactions but I am only seeing 1 transfer of 1 euro rather then both of them.

**Comment by JC5** [comment]
Mmmm. Perhaps it's the new savings things. I just created two equal transactions (moving €5 with the same description) and they're both there.

**Comment by colinerd** [comment]
Just to clarify, I did both transactions of 4 euro’s and Bunq automatically created 2 transfers for 1 euro each (to round up to 5) for the saving goal.
In Firefly, I can see both transacionts of 4 euro but I just see one saving goal transfer of 1 euro while I would expect to also see 2 of them.

I have checked all double saving goal transfers and the amount is exactly the difference I have on the account.
